### PR TITLE
retry logic for setting password

### LINKED ISSUE #94056: Writing login information to the keychain failed with error 'An invalid record was encountered.'.
testing https://github.com/microsoft/vscode/issues/93811

```
Version: 1.44.0-insider
Commit: d7d1147d6b764a48b789000c03088ee9afabcd8a
Date: 2020-03-31T08:22:34.682Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 17.7.0
```

I signed into two accounts (cdias @ microsoft.com and chris @ diasfam.com), and then signed out of the microsoft.com account. The result:

image

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
This PR fixes #94056 by retrying to set the password 5 times.

### CODE REVIEW

File: src/vs/platform/native/electron-main/nativeHostMainService.ts [hunk 1]
```diff
@@ -812,6 +812,27 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	async setPassword(windowId: number | undefined, service: string, account: string, password: string): Promise<void> {
 		const keytar = await this.withKeytar();
+		const MAX_SET_ATTEMPTS = 5;
```
Comment (RMacfarlane) [inline src/vs/platform/native/electron-main/nativeHostMainService.ts:815]: IMO 6 attempts is a bit much, I think if it doesn't succeed after 3 tries, it's probably just not going to succeed. So I suggest making this 2

Comment (RMacfarlane) [reply src/vs/platform/native/electron-main/nativeHostMainService.ts:815]: Especially since that introduces a perceptible wait time

Comment (TylerLeonhardt) [reply src/vs/platform/native/electron-main/nativeHostMainService.ts:815]: Yeah I don't might changing it to 3. I can even shorten the MS to like 200 so it all still operates under 1s.

Comment (TylerLeonhardt) [reply src/vs/platform/native/electron-main/nativeHostMainService.ts:815]: set to 3.

**Review by RMacfarlane — APPROVED** [review]
LGTM!

**Pull request merged**